### ISSUE TITLE
[R-package] Randomness if top_rate and/or other_rate too low

### ISSUE THREAD

**Issue by christopher-rojas** [microsoft/LightGBM#3163]
Hello,

I'm tuning a LightGBM model using its goss version for a multi-class classification problem. However, I'm running into an issue where I observe significant randomness in replication results if top_rate and/or other_rate hyperparameters are too low, even though I thought I had set all of the seeds. Specifically, I have explicitly set the seed for R's random number generator, as well as the bagging_seed and the feature_fraction_seed.

I do not see any other seeds listed in the documentation for lightgbm, but maybe I am missing something. Do you have any ideas if there is an additional seed which needs to be set?

Thanks!

### Environment info
Operating System: UNIX

CPU/GPU model: AWS EC2 r5.24xlarge

C++/Python/R version: rstudio36

LightGBM version or commit hash: 2.2.3

**Comment by jameslamb** [comment]
Thanks for using `LightGBM` and for reporting this issue.

* Can you please share the code you are running, ideally a reproducible example that shoows the issue?
* Can you please share the output of this R command --> `sessionInfo()`.
* How did you install `LightGBM`'s R package? Version 2.2.3 is more than a year old, I'm wondering if you can update to a more recent version.

**Comment by guolinke** [comment]
@christopher-rojas 
As you run the old version, the bagging will be affected by num_thread.
you may need to fix num_thread and try again.
